react/forms/step4.md

```diff
@@ -0,0 +1,71 @@
+PatternFly components, including forms, can be customized with props. You can add a form validation effect to your cards with the `isValid` prop and helper text with the `helperTextInvalid` prop.
+
+## Task
+
+1) First, we'll update our constructor state declaration to have an additional `isValid` state prop:
+
+<pre class="file" data-target="clipboard">
+constructor(props) {
+  super(props);
+  this.state = {
+    value1: '',

```

Comment:
This wipes out the previous value:

`value1: 'please choose',`

It also only partially copies the constructor -- the closing bracket is missing

```
constructor(props) {
  super(props);
  this.state = {
    value1: '',
    value2: '',
    value3: '',
    value4: '',
    isValid: false
  };
```